subgraphs/stablejoe/src/mappings/stablejoe.ts

```diff
@@ -0,0 +1,186 @@
+  }
+  user.updatedAt = event.block.timestamp
+  
+  stableJoe.joeStaked = stableJoe.joeStaked.plus(convertAmountToDecimal(event.params.amount))
+  stableJoe.joeStakedUSD = stableJoe.joeStakedUSD.plus(convertAmountToDecimal(event.params.amount).times(getJoePrice()))
+  stableJoe.updatedAt = event.block.timestamp
+
+  stableJoe.save()
+
+  // update deposit fees 
+  if (stableJoe.depositFee) {
+    stableJoe.depositFeeJOE = stableJoe.depositFeeJOE.plus(stableJoe.depositFee)
+    stableJoe.depositFeeUSD = stableJoe.depositFeeUSD.plus(stableJoe.depositFee).times(getJoePrice())
+  }
+
+  // update day data
+  const timestamp = event.block.timestamp.toI32()
+  const day = timestamp / 86400
+  const date = day * 86400
+  const id = event.address.toHex().concat('-').concat(BigInt.fromI32(day).toString())
+
+  let stableJoeDayData = StableJoeDayData.load(id);
+  if (stableJoeDayData === null) {
+    stableJoeDayData = new StableJoeDayData(id)
+    stableJoeDayData.date = date
+    stableJoeDayData.joeStaked = convertAmountToDecimal(event.params.amount)
+    stableJoeDayData.joeStakedUSD = convertAmountToDecimal(event.params.amount).times(getJoePrice())
+    if (stableJoe.depositFee) {
+      stableJoeDayData.depositFeeJOE = stableJoe.depositFee
+      stableJoeDayData.depositFeeUSD = stableJoe.depositFee.times(getJoePrice())
+    }
+  } else {
+    stableJoeDayData.joeStaked = stableJoeDayData.joeStaked.plus(convertAmountToDecimal(event.params.amount))
+    stableJoeDayData.joeStakedUSD = stableJoeDayData.joeStakedUSD.plus(convertAmountToDecimal(event.params.amount).times(getJoePrice()))
+    if (stableJoe.depositFee) {
+      stableJoe.depositFeeJOE = stableJoe.depositFeeJOE.plus(stableJoe.depositFee)
+      stableJoe.depositFeeUSD = stableJoe.depositFeeUSD.plus(stableJoe.depositFee).times(getJoePrice())
+    }
+  }
+
+  stableJoeDayData.save()
+}
+
+export function handleDepositFeeChanged(event: DepositFeeChangedEvent): void {
+  const stableJoe = StableJoe.load(event.address.toHex())
+
+  stableJoe.depositFee = convertAmountToDecimal(event.params.newFee)
+  stableJoe.updatedAt = event.block.timestamp
+  stableJoe.save()
+}
+
+export function handleWithdraw(event: WithdrawEvent): void {
+  const stableJoe = StableJoe.load(event.address.toHex())
+  let user = User.load(event.params.user.toHex())
+
+  if (!user) return
+
+  user.joeStaked = user.joeStaked.minus(convertAmountToDecimal(event.params.amount))
+  user.joeStakedUSD = user.joeStakedUSD.minus(convertAmountToDecimal(event.params.amount).times(getJoePrice()))
+  user.updatedAt = event.block.timestamp
+
+  const joePrice = getJoePrice()
+  stableJoe.joeStaked = stableJoe.joeStaked.minus(convertAmountToDecimal(event.params.amount))
+  stableJoe.joeStakedUSD = stableJoe.joeStakedUSD.minus(convertAmountToDecimal(event.params.amount).times(getJoePrice()))
+  stableJoe.updatedAt = event.block.timestamp
+
+  stableJoe.save()
+
+  // update day data
```

Comment:
We don't want to keep track of unstaked here so are we just going to skip updating the day data in the case of withdraws? Since it would be making joeStaked negative if this was the first event of that "day". I think we decided to skip but forgot what we concluded @0xmurloc

Reply:
joeStaked is a current balance, so on withdraw we should deduct the amount. 

something like `joeStaked = withdrawAmount.gt(joeStaked) ? 0 : joeStaked.sub(withdrawAmount)`